### PR TITLE
1

### PR DESCRIPTION
**Pull request by serhii-dvorets**
https://serhii-dvorets.github.io/react_goods-selector/

### CODE REVIEW

**Review by danylo-onishchuk — CHANGES_REQUESTED** [review]
Add button X for to the h1 to clear the selection. (Set empty string '')

File: src/App.tsx [hunk 1]
```diff
@@ -14,11 +15,98 @@ const goodsFromServer: string[] = [
...
+    selectedGoods: [],
+  };
+
+  remover = (index: number, array: string[]) => {
+    array.splice(index, 1);
+
+    return array;
+  };
+
+  makephrase = (array: string[]) => {
```
Comment (aliceblunt8) [inline src/App.tsx:35]:
```suggestion
  selectedGoodsTitle = (array: string[]) => {
```

File: src/App.tsx [hunk 2]
```diff
@@ -14,11 +15,98 @@ const goodsFromServer: string[] = [
...
+            onClick={() => {
+              this.setState({ selectedGoods: [] });
+            }}
+          >
+            X
+          </button>
+        </div>
+        {
+          goodsFromServer.map((item) => (
+            <p>
```
Comment (aliceblunt8) [inline src/App.tsx:77]: why do you need p?

File: src/App.tsx [hunk 3]
```diff
@@ -14,11 +15,98 @@ const goodsFromServer: string[] = [
...
+            className="clearButton"
+            onClick={() => {
+              this.setState({ selectedGoods: [] });
+            }}
+          >
+            X
+          </button>
+        </div>
+        {
+          goodsFromServer.map((item) => (
```
Comment (aliceblunt8) [inline src/App.tsx:76]: add key

File: src/App.tsx [hunk 4]
```diff
@@ -14,11 +15,98 @@ const goodsFromServer: string[] = [
...
+                        selectedGoods: [...prevState.selectedGoods, item],
+                      }));
+                    } else {
+                      const index = selectedGoods.indexOf(item);
+
+                      this.setState((prevState: State) => ({
+                        selectedGoods: this.remover(index, prevState.selectedGoods),
+                      }));
+                    }
+                  }}
```
Comment (aliceblunt8) [inline src/App.tsx:97]: move to function

**Review by aliceblunt8 — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 3 above)
Comment (serhii-dvorets) [reply src/App.tsx:76]: What key?

**Review by danylo-onishchuk — APPROVED** [review]

**Pull request closed without merging**